go-controller/pkg/libovsdbops/router.go

```diff
@@ -351,21 +351,25 @@ func addOrUpdateNATToRouterOps(nbClient libovsdbclient.Client, ops []libovsdb.Op
 }
 
 func AddOrUpdateNATsToRouterOps(nbClient libovsdbclient.Client, ops []libovsdb.Operation, router *nbdb.LogicalRouter, nats ...*nbdb.NAT) ([]libovsdb.Operation, error) {
+	var routerName string
+	if router != nil {
```

Comment:
do we need this check? do we have a case where we are calling this function with a router object where name isn't set? Why can't we just use router.Name directly like in `AddLoadBalancersToRouterOps`?

Reply:
the router != nil check isn't required as long as we guarantee we never pass a nil object. I don't think we do that today anyway because the functions that get called inside of AddOrUpdate also assume its not nil. I could remove it, but I don't think it hurts.

The reason I need to use a routerName variable is because I think findRouter for example will return a nil router object when there is an error, and then my error log would cause an NPE using router.Name

Reply:
> the router != nil check isn't required as long as we guarantee we never pass a nil object. I don't think we do that today anyway because the functions that get called inside of AddOrUpdate also assume its not nil. I could remove it, but I don't think it hurts.
> 

ack yeah `routerName := router.Name` would be cleaner but I don't think your check hurts anyways just that its redundant at this point like you said since we don't really do anything if its nil. 

> The reason I need to use a routerName variable is because I think findRouter for example will return a nil router object when there is an error, and then my error log would cause an NPE using router.Name

ah yeah sorry I didn't realize we were refilling/resuing the same pointer object.